core/capabilities/ccip/oraclecreator/plugin.go

```diff
@@ -266,98 +246,234 @@ func (i *pluginOracleCreator) createReadersAndWriters(
+		}
+		ofc.commitOffchainConfig = &commitOffchainCfg
+	}
+	if !ofc.isValid() {
+		return offChainConfig{}, fmt.Errorf("invalid offchain config: both commit and exec configs are either set or unset")
+	}
+	return ofc, nil
+}
+
+func (i *pluginOracleCreator) getChainSelector(chainID uint64) (cciptypes.ChainSelector, error) {
+	chainSelector, ok := chainsel.EvmChainIdToChainSelector()[chainID]
+	if !ok {
+		return 0, fmt.Errorf("failed to get chain selector from chain ID %d", chainID)
+	}
+	return cciptypes.ChainSelector(chainSelector), nil
+}
+
+func (i *pluginOracleCreator) getChainReaderConfig(
+	chainID uint64,
+	destChainID uint64,
+	ofc offChainConfig,
+	chainSelector cciptypes.ChainSelector,
+) evmrelaytypes.ChainReaderConfig {
+	var chainReaderConfig evmrelaytypes.ChainReaderConfig
+	if chainID == destChainID {
+		chainReaderConfig = evmconfig.DestReaderConfig
+	} else {
+		chainReaderConfig = evmconfig.SourceReaderConfig
+	}
+
+	if !ofc.commitEmpty() && ofc.commit().PriceFeedChainSelector == chainSelector {
+		chainReaderConfig = evmconfig.MergeReaderConfigs(chainReaderConfig, evmconfig.FeedReaderConfig)
+	}
+	return chainReaderConfig
+}
+
+func (i *pluginOracleCreator) createChainReader(
+	chain legacyevm.Chain,
+	chainReaderConfig evmrelaytypes.ChainReaderConfig,
+	pluginType cctypes.PluginType,
+) (types.ContractReader, error) {
+	cr, err := evm.NewChainReaderService(
+		context.Background(),
+		i.lggr.
+			Named("EVMChainReaderService").
+			Named(chain.ID().String()).
+			Named(pluginType.String()),
+		chain.LogPoller(),
+		chain.HeadTracker(),
+		chain.Client(),
+		chainReaderConfig,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create contract reader for chain %s: %w", chain.ID(), err)
+	}
+	return cr, nil
+}
+
+func (i *pluginOracleCreator) bindContracts(
+	chain legacyevm.Chain,
+	cr types.ContractReader,
+	config cctypes.OCR3ConfigWithMeta,
+	destChainID uint64,
+	ofc offChainConfig,
+	chainSelector cciptypes.ChainSelector,
+) error {
+	if chain.ID().Uint64() == destChainID {
+		offrampAddressHex := common.BytesToAddress(config.Config.OfframpAddress).Hex()
+		err := cr.Bind(context.Background(), []types.BoundContract{
+			{
+				Address: offrampAddressHex,
+				Name:    consts.ContractNameOffRamp,
+			},
+		})
+		if err != nil {
+			return fmt.Errorf("failed to bind chain reader for dest chain %s's offramp at %s: %w", chain.ID(), offrampAddressHex, err)
+		}
+	}
 
-		chainSelector, ok := chainsel.EvmChainIdToChainSelector()[chain.ID().Uint64()]
-		if !ok {
-			return nil, nil, fmt.Errorf("failed to get chain selector from chain ID %s", chain.ID())
+	if !ofc.commitEmpty() && chainSelector == ofc.commit().PriceFeedChainSelector {
+		tokenInfo := ofc.commit().TokenInfo
+		var bcs []types.BoundContract
+		for _, info := range tokenInfo {
+			bcs = append(bcs, types.BoundContract{
+				Address: info.AggregatorAddress,
+				Name:    consts.ContractNamePriceAggregator,
+			})
+		}
```

Comment:
I feel like we will run into the same issue we did with the offramps, these will have to be re-bound in the chainlink-ccip code because we call `GetBindings` (I think)

Reply:
Since these won't be used for events and just calls, and since we have all the addresses anyway in the plugin, I'm tempted to just do this in the plugin codebase instead and simplify the startup code.

Reply:
Why do we bind offramp here then? We bind it again in chainlink-ccip, right?

Reply:
Yeah, I think we should revisit this.